### PR TITLE
Remove duplicate `FidesUpdated` event and trigger `FidesInitialized` twice instead

### PR DESCRIPTION
**Pull request by NevilleS**
Closes [PROD-1280](https://ethyca.atlassian.net/browse/PROD-1280)

### Description Of Changes

`FidesUpdated` is now only called when a user makes a change, not after initialization. 

https://github.com/ethyca/fides/assets/24641006/cc23f278-9f79-450e-a6cd-c45b87b6499c

### Code Changes

* [x] Remove calls to `FidesUpdated` in initialization logic
* [x] Simplify `FidesInitialized` calls
* [x] Update tests

### Steps to Confirm

* [ ] Set up TCF or notices
* [ ] Visit the demo page and open the console
* [ ] You should see that `FidesInitialized` fired but no `FidesUpdated` fired
* [ ] You should not see `FidesUpdated` until you save your preferences

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

[PROD-1280]: https://ethyca.atlassian.net/browse/PROD-1280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/fides-js/src/fides-tcf.ts [hunk 1]
```diff
@@ -225,14 +223,9 @@ const init = async (config: FidesConfig) => {
...
-  // Dispatch the "FidesInitialized" event to update listeners with the initial
-  // state. Skip if we already initialized due to an existing cookie.
-  // For convenience, also dispatch the "FidesUpdated" event; this allows
-  // listeners to ignore the initialization event if they prefer
-  if (isNewFidesCookie(cookie)) {
-    dispatchFidesEvent("FidesInitialized", cookie, config.options.debug);
-  }
-  dispatchFidesEvent("FidesUpdated", cookie, config.options.debug);
+  // Dispatch the "FidesInitialized" event to update listeners with the initial state.
+  // TODO PROD-1280: Suppress duplicate event if the cookie state is unchanged from an early initialization?
```
Comment (NevilleS) [inline clients/fides-js/src/fides-tcf.ts:227]: I'm really not sure about this yet. I like the _simplicity_ of this code right now, but the double-triggering of the event doesn't "feel" great. Looking for opinions 😄

File: clients/privacy-center/cypress/e2e/consent-banner.cy.ts [hunk 2]
```diff
@@ -1438,27 +1409,15 @@ describe("Consent banner", () => {
...
-          .should("have.been.calledOnce")
-          .its("firstCall.args.0.detail.consent")
-          .should("deep.equal", {
-            data_sales: false,
-            tracking: false,
-            analytics: true,
-          });
-        cy.get("@FidesUpdated")
+          // TODO PROD-1280: Suppress duplicate event if the cookie state is unchanged from an early initialization?
+          // .should("have.been.calledOnce")
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/consent-banner.cy.ts:1413]: Note that we shouldn't check this in; need to decide what to do and then do it 👍

File: clients/fides-js/src/fides.ts [hunk 3]
```diff
@@ -126,14 +124,9 @@ const init = async (config: FidesConfig) => {
...
-  // Dispatch the "FidesInitialized" event to update listeners with the initial
-  // state. Skip if we already initialized due to an existing cookie.
-  // For convenience, also dispatch the "FidesUpdated" event; this allows
-  // listeners to ignore the initialization event if they prefer
-  if (isNewFidesCookie(cookie)) {
-    dispatchFidesEvent("FidesInitialized", cookie, config.options.debug);
-  }
-  dispatchFidesEvent("FidesUpdated", cookie, config.options.debug);
+  // Dispatch the "FidesInitialized" event to update listeners with the initial state.
+  // TODO PROD-1280: Suppress duplicate event if the cookie state is unchanged from an early initialization?
```
Comment (NevilleS) [inline clients/fides-js/src/fides.ts:128]: TODO note 👍

**Review by NevilleS — COMMENTED** [review]
Comments and thoughts

**Comment by allisonking** [comment]
We decided to move forward with this implementation for now, but perhaps we should suppress the duplicate `FidesInitialized` during our refactor of our initialization logic. Right now it's just complicated to figure out if the initialization should fire again.

**Review by eastandwestwind — APPROVED** [review]

**Pull request merged**